### DISCUSSION TITLE
More information about PnP Modern Search API Access permissions

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2381] [Q&A] [answered]
Hi Team,
We are evaluating this PnP Modern Search solution at QA environment and would like to understand more about the below concerns.
1. Would like understand how Web Api access permissions will work at backend with Microsoft Graph API ? like the App will consume  delegated or APP permissions ?
2. If delegated (As I know ) this will consume based the current logged in user and based on the user only the can access the tenant user only data, if it is APP permissions? , please explain.

Please give more information about Web Api access permissions and the way how to consume in with in the PnP Solution.
thank you,

**Comment by wobba** [comment, marked as answer]
1. Read the documentation at https://learn.microsoft.com/en-us/sharepoint/dev/spfx/use-aad-tutorial for how SPFx works with granting Graph scopes in SPFx. But yes, they are delegated with user_impersonation.
2. See 1. And also see https://microsoft-search.github.io/pnp-modern-search/installation/#installation which explains you only need to grant scopes you will need, not all requested ones.
